atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -350,34 +430,53 @@ public void hostsAreNotRemovedOrAddedWhenRefreshIsDisabled() {
 
         setCassandraServersTo(CASS_SERVER_1);
         createClientPool();
-        assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
+        assertThat(poolServers).containsOnlyKeys(CASS_SERVER_1, CASS_SERVER_2);
 
         setCassandraServersTo(CASS_SERVER_1, CASS_SERVER_2, CASS_SERVER_3);
-        deterministicExecutor.tick(POOL_REFRESH_INTERVAL_SECONDS, TimeUnit.SECONDS);
-        assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
+        refreshPool();
+        assertThat(poolServers).containsOnlyKeys(CASS_SERVER_1, CASS_SERVER_2);
+    }
+
+    @Test
+    public void throwsWhenTryingToAddServerPresentInCurrentServers() {
+        setupThriftServers(ImmutableSet.of(CASS_SERVER_1.proxy()));
+        when(config.autoRefreshNodes()).thenReturn(true);
+
+        setCassandraServersTo(CASS_SERVER_1, CASS_SERVER_2);
+
+        CassandraClientPoolImpl pool = createClientPool();
+        assertThat(poolServers).containsOnlyKeys(CASS_SERVER_1, CASS_SERVER_2);
+        assertThatThrownBy(() -> pool.validateNewHostsTopologiesAndMaybeAddToPool(poolServers, Set.of(CASS_SERVER_1)))
+                .isInstanceOf(IllegalArgumentException.class)
```

Comment:
consider: This is the exception thrown by the immutable map (IIRC) - since it's not a safe log, we won't see the message nor the args. 

Consider doing a precondition check to see if there are any duplicates and explicitly safe arg (since they're servers) so there's more debugging information